lib/Request.js

```diff
@@ -310,3 +313,89 @@ Request.prototype.propertyError = function(property, errorCode, httpStatusCode,
         });
     }.bind(this));
 };
+
+Request.prototype.getStream = function() {
+    if (null === this._stream) {
+        this._stream = new Busboy({
+            headers: this._connection.raw.req.headers
+        });
+        this._connection.raw.req.pipe(this._stream);
+    }
+    return this._stream;
+}
+
+Request.prototype.getUpload = function(field) {
+    return new Promise(function(resolve, reject) {
+        var stream = this.getStream();
+        var found = false;
+
+        stream.on('file', function(uploadedFieldName, file, filename, encoding, mimetype) {
+            if (field != uploadedFieldName) {
+                // Got an upload, but not for the field name we are looking for.
+                // Skip this upload, maybe the next one will be under the desired field name.
+                return;
+            }
+
+            found = true;
```

Comment:
Does this work? What if the first field matches, the second one doesn't, and the third does? Do all three get handled here / in stream.on('finish' ?